tests/test_sa_engine.py

```diff
@@ -80,10 +81,10 @@ def test_not_context_manager(engine):
 async def test_release_transacted(engine):
     conn = await engine.acquire()
     tr = await conn.begin()
-    with pytest.raises(sa.InvalidRequestError):
-        engine.release(conn)
+    with pytest.warns(ResourceWarning, match='Invalid transaction status'):
+        await engine.release(conn)
```

Comment:
Maybe it is better to raise an exception here instead of a warning. Looks like a user's bug to release a connection in the middle of transaction.

Reply:
I've added this comment to the [issue](https://github.com/aio-libs/aiopg/issues/761) as well.